test/modules/TestResultCoordinator/TestReportUtil.cs

```diff
@@ -103,9 +110,100 @@ internal static List<ITestReportMetadata> ParseReportMetadataJson(string reportM
             return reportMetadataList;
         }
 
+        internal static async Task<Uri> GetOrCreateBlobContainerSasUriForLogAsync(string storageAccountConnectionString)
```

Comment:
Can we move all upload logic to a new class? This one will get bloated, and TestReportUtil is already a generic name